typescript/sdk/src/metadata/ChainMetadataManager.ts

```diff
@@ -210,9 +213,16 @@ export class ChainMetadataManager<MetaExt = {}> {
    * Get a block explorer URL for a given chain name, chain id, or domain id
    */
   tryGetExplorerUrl(chainNameOrId: ChainName | number): string | null {
-    const explorers = this.tryGetChainMetadata(chainNameOrId)?.blockExplorers;
-    if (!explorers?.length) return null;
-    return explorers[0].url;
+    const metadata = this.tryGetChainMetadata(chainNameOrId);
+    if (!metadata?.blockExplorers?.length) return null;
+    const url = new URL(metadata.blockExplorers[0].url);
+    if (
+      metadata.protocol === ProtocolType.Sealevel &&
+      solanaChainToClusterName[metadata.name]
+    ) {
+      url.searchParams.set('cluster', solanaChainToClusterName[metadata.name]);
+    }
+    return url.toString();
```

Comment:
It could be, we could have a generic `queryParams` property on the ChainMetadata interface.
Tbh this was me cutting a corner to get this hack brought over from the wrap UI.

Reply:
Left a TODO